### PR TITLE
Add support for custom themes and providers

### PR DESCRIPTION
**Pull request by TheBigLee**
## Summary

* This PR adds support for custom themes and providers

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [ ] Update tests.
- [ ] Link this PR to related issues.

### CODE REVIEW

File: apis/vshn/v1/dbaas_vshn_keycloak.go [hunk 1]
```diff
@@ -100,6 +101,19 @@ type VSHNKeycloakServiceSpec struct {
 	// PostgreSQLParameters can be used to set any supported setting in the
 	// underlying PostgreSQL instance.
 	PostgreSQLParameters *VSHNPostgreSQLParameters `json:"postgreSQLParameters,omitempty"`
+
+	// CustomThemeProviderImage can be used to provide an image with custom themes and providers.
+	// The themes need to be be placed in the `/themes` directory of the custom image.
+	// the providers need to be placed in the `/providers` directory of the custom image.
+	CustomThemeProvider VSHNKeycloakCustomThemeProvider `json:"customThemeProvider,omitempty"`
```
Comment (Kidswiss) [inline apis/vshn/v1/dbaas_vshn_keycloak.go:108]: Maybe another name for this? The comments imply that this thing handles providers and themes, naming is hard and the variable sounds like it only provides themes.

File: pkg/comp-functions/functions/vshnkeycloak/deploy.go [hunk 2]
```diff
@@ -412,6 +508,26 @@ func toYAML(obj any) (string, error) {
 	return string(yamlBytes), err
 }
 
+func copyCustomImagePullSecret(svc *runtime.ServiceRuntime, comp *vshnv1.VSHNKeycloak) error {
+	secretClaim := &corev1.Secret{}
+
+	err := svc.SetDesiredKubeObserveObject(secretClaim, comp.GetName()+"-custom-image-pull-secret-observer")
+	if err != nil {
+		return err
+	}
```
Comment (Kidswiss) [inline pkg/comp-functions/functions/vshnkeycloak/deploy.go:517]: You don't seem to get this observer ever again... So it's probably not necessary?

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Some inputs

File: apis/vshn/v1/dbaas_vshn_keycloak.go (hunk 1 above)
Comment (TheBigLee) [reply apis/vshn/v1/dbaas_vshn_keycloak.go]: Changed it to something more meaningful

File: pkg/comp-functions/functions/vshnkeycloak/deploy.go (hunk 2 above)
Comment (TheBigLee) [reply pkg/comp-functions/functions/vshnkeycloak/deploy.go]: Fixed

File: pkg/comp-functions/functions/vshnkeycloak/deploy.go [hunk 3]
```diff
@@ -449,7 +482,35 @@ func addPullSecret(comp *vshnv1.VSHNKeycloak, svc *runtime.ServiceRuntime) error
...
+		customImagePullSecret = map[string]any{
+			"name": customImagePullsecretName,
+		}
+	}
+
+	pullSecrets := []map[string]any{
+		{
+			"name": pullsecretName,
+		},
+		customImagePullSecret,
```
Comment (Kidswiss) [inline pkg/comp-functions/functions/vshnkeycloak/deploy.go:502]: What happens if no pull secret is specified in the claim? Does it add an invalid secret and break everything?

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Just one thing to clarify.

File: pkg/comp-functions/functions/vshnkeycloak/deploy.go (hunk 3 above)
Comment (TheBigLee) [reply pkg/comp-functions/functions/vshnkeycloak/deploy.go:502]: The field would look like this:
```
imagePullSecrets:
- name: pullsecret
- {}
```

It's not pretty. but it works

Comment (TheBigLee) [reply pkg/comp-functions/functions/vshnkeycloak/deploy.go]: I've refactored that part, to not generate an empty `{}` if a pullSecret is not specified.

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**